plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -79,4 +82,25 @@ public void testShowCreateTable()
                         format("   location = '.*/" + schemaName + "/region'\n") +
                         "\\)");
     }
+
+    @Test
+    public void testHiddenPathColumn()
+    {
+        String tableName = "hidden_file_column_table_metastore_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a int, b VARCHAR) ");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 'a')", 1);
+        assertQuery("SELECT a, b FROM " + tableName, "VALUES (1, 'a')");
+
+        MaterializedResult result = computeActual(format("SELECT file_path FROM \"%s$files\"", tableName));
+        assertEquals(result.getRowCount(), 1);
+        String filePath = (String) result.getOnlyValue();
+
+        assertQuery("SELECT DISTINCT \"$path\" FROM " + tableName, "VALUES " + "'" + filePath + "'");
+
+        // Check whether the "$path" hidden column is correctly evaluated in the filter expression
+        assertQuery(format("SELECT a,b FROM %s WHERE \"$path\" = '%s'", tableName, filePath), "VALUES (1, 'a')");
+
+        assertUpdate("DROP TABLE " + tableName);
```

Comment:
It would be better to use `TestTable` considering test failure on the way. Also, there is `computeScalar()` method. 
```java
        try (TestTable table = new TestTable(getQueryRunner()::execute, "hidden_path_column", "(a int, b VARCHAR)", ImmutableList.of("(1, 'a')"))) {
            String filePath = (String) computeScalar(format("SELECT file_path FROM \"%s$files\"", table.getName()));

            assertQuery("SELECT DISTINCT \"$path\" FROM " + table.getName(), "VALUES " + "'" + filePath + "'");

            // Check whether the "$path" hidden column is correctly evaluated in the filter expression
            assertQuery(format("SELECT a FROM %s WHERE \"$path\" = '%s'", table.getName(), filePath), "VALUES 1");
        }
```

Reply:
thanks @ebyhr

Reply:
update @ebyhr